packages/wallets/src/features/cashier/modules/Transfer/hooks/useModifyTransferAccounts.ts

```diff
@@ -0,0 +1,187 @@
+import { useMemo } from 'react';
+import { useActiveWalletAccount, useAuthorize, useCurrencyConfig } from '@deriv/api';
+import { displayMoney } from '@deriv/api/src/utils';
+import { THooks } from '../../../../../types';
+import { getAccountName, getLandingCompanyNameOfMT5Account, getMarketType } from '../helpers';
+
+type TAccountType = THooks.TransferAccount['account_type'];
+type TMt5MarketType = ReturnType<typeof getMarketType>;
+
+/** A custom hook that enhances the transfer accounts response by adding additional properties for convenient UI rendering. */
+const useModifyTransferAccounts = (accounts?: THooks.TransferAccount[]) => {
+    const { data: authorizeData, isLoading: isAuthorizeLoading } = useAuthorize();
+    const { data: activeWallet, isLoading: isActiveWalletLoading } = useActiveWalletAccount();
+    const { getConfig, isLoading: isCurrencyConfigLoading } = useCurrencyConfig();
+
+    const isLoading = isAuthorizeLoading || isCurrencyConfigLoading || isActiveWalletLoading;
+
+    const modifiedTransferAccounts = useMemo(() => {
+        //populating transfer accounts with extra fields
+        const updatedAccounts = accounts?.map(account => {
+            const currencyConfig = account.currency ? getConfig(account.currency) : undefined;
+            const accountName = getAccountName({
+                accountCategory: account.account_category,
+                accountType: account.account_type,
+                displayCurrencyCode: currencyConfig?.display_code,
+                landingCompanyName: activeWallet?.landing_company_name,
+                mt5MarketType: getMarketType(account.mt5_group),
+            });
+            const activeWalletIcon = activeWallet?.is_virtual
+                ? activeWalletIconMapper.Demo?.light
+                : activeWalletIconMapper[activeWallet?.currency as keyof typeof activeWalletIconMapper]?.light;
+            const appIcon =
```

Comment:
I just realised that we are controlling what icons to display from here as well. I think deciding what icons to show in a hook is an `anti-pattern`. Can you refactor this out of the hook and create a utility function that handles all this logic of deciding which icon to show and use it in the component instead?

Reply:
agree with that, it have been discussed quite a couple times in comments, that visual concerns should be separated from hooks, as hooks in our case are used to provide data,

Reply:
@yashim-deriv 
okey guys, i extracted `activeWalletIcon` and `appIcon` to helper function.
i have question regarding the rest of extra props that i added here:  accountName, currencyConfig, displayBalance and landingCompanyName. Is it fine to leave them here?

Reply:
for me its about separation of concerns, 
in my mind, activeName, currencyConfig, landingCompanyName do belong to account, so its fair to keep them in hook with account. In a way, those are properties of account (ideally I guess we should have clearly defined account model, but in absence of those, common sense says, that it should be used that way)

displayBalance feels a bit more like a component responsibilty, to display balance properly based on currencyConfig and balance itself, but I do not mind if it stays, its very much arguable where it should be, 

so looks good to me,

Reply:
got it @wojciech-deriv thanks for clarification